### PR TITLE
Add [ QA | ARCHIVER | REPORT ]

### PR DESCRIPTION
**Pull request by nickumia-reisys**
Related to
- https://github.com/GSA/data.gov/issues/3957
- https://github.com/ckan/ckanext-archiver/pull/73
- https://github.com/ckan/ckanext-qa/pull/74
- https://github.com/ckan/ckanext-report/pull/25

### CODE REVIEW

File: ckan/requirements.in [hunk 1]
```diff
@@ -6,6 +6,9 @@ ckanext-dcat @ git+https://github.com/ckan/ckanext-dcat@master
 -e git+https://github.com/gsa/ckanext-spatial.git@datagov#egg=ckanext-spatial
 # git+https://github.com/keitaroinc/ckanext-saml2auth.git@main#egg=ckanext-saml2auth
 git+https://github.com/gsa/ckanext-saml2auth.git@main#egg=ckanext-saml2auth
+-e git+https://github.com/gsa/ckanext-qa.git@datagov#egg=ckanext-qa
+-e git+https://github.com/ckan/ckanext-archiver.git@master#egg=ckanext-archiver
+-e git+https://github.com/ckan/ckanext-report.git@master#egg=ckanext-report
```
Comment (jbrown-xentity) [inline ckan/requirements.in:11]: Just to make sure, do these *need* to be installed in editable mode? Did we try without? Either way, could we make a comment (either not tested, or with a note about errors when installing without `-e`?

File: ckan/setup/GSA_prerun.py [hunk 2]
```diff
@@ -14,6 +16,25 @@
...
+def init_db():
+
+    db_command = ["ckan", "-c", pr.ckan_ini, "db", "init"]
+    print("[prerun] Initializing or upgrading db - start")
+    try:
+        subprocess.check_output(db_command, stderr=subprocess.STDOUT)
+        print("[prerun] Initializing or upgrading db - end")
+    except subprocess.CalledProcessError as e:
+        # GSA FIX: e.output is bytes upstream
+        if "OperationalError" in str(e.output):
```
Comment (jbrown-xentity) [inline ckan/setup/GSA_prerun.py:28]: Can we push this fix upstream?

**Review by jbrown-xentity — APPROVED** [review]
Looks good, and could be merged as is. Thanks @nickumia-reisys !

File: ckan/requirements.in (hunk 1 above)
Comment (nickumia-reisys) [reply ckan/requirements.in:11]: If it's not in editable mode, the custom templates are missing and can't be accessed.  So yeah, we need it in editable mode.  Also, the cli commands don't work if not editable.

(I did try it without editable initially)

File: ckan/setup/GSA_prerun.py (hunk 2 above)
Comment (nickumia-reisys) [reply ckan/setup/GSA_prerun.py:28]: I'm not sure.  I think this is a PY2 vs PY3 issue that upstream is still allowing support for PY2 😕

**Pull request merged**